### PR TITLE
Build with depend

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Removed local download/clone from scripts/build.sh and packaging/ios/build.sh
and replaced with get-dep-lib

Android is not touched, because I currently do not have a android test environment
(soon to be changed).

Feel free to make changes to my commits.

### CODE REVIEW

**Comment by dirkhh** [comment]
It looks like something is off with libgit2 - Mac, Linux, and Linux/qt55 are failing because of it.
I haven't looked into "why" at all. Happy to help, but I'm assuming that you have started to look into it already?

**Comment by dirkhh** [comment]
Yes, Travis isn't perfect, but it sure helps a lot.
I'll simply wait for you here - this isn't urgent.

**Comment by janiversen** [comment]
Do not ask me how, but when I look at the raw log of the failing jobs, I cannot find the call to get-dep-lib (even though e.g. my pushd commands are there). Can you help me explain that ?

**Comment by janiversen** [comment]
I think I got it, can it be travis runs without -build-deps ? 

That would explain why I do not see the get-dep-lib call.

**Comment by dirkhh** [comment]
Travis is controlled by https://github.com/Subsurface-divelog/subsurface/blob/master/.travis.yml
which calls three scripts under `scripts/${SUBSURFACE_PLATFORM}/ ` depending on which of the test environments it is. Those scripts then trigger the setup, the actual build, and what to do after the build succeeds.
Right now, for Mac, we switched to not building our dependencies and instead using them from Homebrew (because it ended up that that was much faster). The call to `build.sh` is here https://github.com/Subsurface-divelog/subsurface/blob/master/scripts/mac/travisbuild.sh#L18

**Comment by dirkhh** [comment]
Oh, and the reason this is faster is that we have a setup that allows us to run Homebrew to build everything and then cache that - and the download of the cache is much faster than the build.
Obviously, the build needs to work in either of these two scenarios.
Equally obviously (I hope), for your testing you could simply modify Line 18 referenced in the previous comment and have it build things as you want to see them built.

**Comment by janiversen** [comment]
Finally ! will integrate in the android build once this is done.

**Review by dirkhh — COMMENTED** [review]
Hmm - half way through the review the commits changed again... I'll submit this review now and start again with the latest versions, I guess...

File: scripts/build.sh [hunk 1]
```diff
@@ -193,63 +193,70 @@ if [[ $PLATFORM = Darwin || "$LIBGIT" < "24" ]] ; then
 	# feature). So we painfully need to build the dependencies ourselves.
 
 	if [ "$BUILD_DEPS" == "1" ] ; then
```
Comment (dirkhh) [inline scripts/build.sh:195]: so all of the changes in this commit are inside the `build-deps` condition. The commit message sounds like you are making changes for Travis, but of course this code isn't run on Travis.
I guess I'm confused...

**Review by dirkhh — CHANGES_REQUESTED** [review]
I really like the direction you are taking this.
The central "getter" script is definitely an improvement.
But some of the changes in this patch series confuse me... I think this may need a little more work (at least to make the commit messages and actual changes correlate better)

**Comment by janiversen** [comment]
I will implement your suggestions, and update the commit messages.

Right now I ran into a "naughty" problem. curl is accepted on all platforms except Android, do I really have to make a "if android use wget else use curl" ?

**Comment by dirkhh** [comment]
We should be able to just install curl inside the container (note, Travis builds the Android apk inside a Ubuntu container, not inside the "outside VM" - so you need to install it as part of the `android-wrapper-script`, IIRC.

**Comment by janiversen** [comment]
Cleaned up commits (the 2 commits earlier of scripts/build.sh were 2 different ideas) and texts.

Also implemented your comments.

Let us see, how much I have broken :-)

Will do the android part, once the general parts works.

**Review by dirkhh — APPROVED** [review]

**Comment by dirkhh** [comment]
Android fails when building googlemaps plugin with 
```
/workspace/subsurface/packaging/android/build.sh: line 165: QMAKE: unbound variable
```

**Comment by janiversen** [comment]
Already corrected, it was my semi-manual merge that took out too many lines (me blames my dm, because I am in parallel analysing my dives from earlier today)

**Comment by janiversen** [comment]
This is not sufficient, how about I just quit the android integration. The rest should be ready to merge.

I can then try to setup a android build environment and test locally (but that will take some days)

wdyt?

**Comment by dirkhh** [comment]
Works for me

**Comment by dirkhh** [comment]
The nice thing about the way Android works on Travis is that it's really easy to recreate the exact same environment locally - simply install Ubuntu in a VM, install the packages listed and then run the scripts...

**Comment by janiversen** [comment]
Yeah travis is GREEEN :-)

I am downloading ubuntu right now, but I suspect it will take a little while to get things working. I already have a android vm for testing. So the good thing (for the project) is that I should be able to e.g. try and bump the library versions myself.

**Pull request merged**